### PR TITLE
Handling Float Integrals

### LINKED ISSUE #14431: integrate((x-t)**(-1/2)*t,(t,0,x)) raises ValueError
I ran this in the current master branch and I got following `ValueError`

```
In [1]: integrate((x-t)**(-1/2)*t,(t,0,x))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-72a531799a51> in <module>()
----> 1 integrate((x-t)**(-1/2)*t,(t,0,x))

F:\Git\sympy\sympy\integrals\integrals.py in integrate(*args, **kwargs)
   1388     if isinstance(integral, Integral):
   1389         return integral.doit(deep=False, meijerg=meijerg, conds=conds,
-> 1390                              risch=risch, manual=manual)
   1391     else:
   1392         return integral

F:\Git\sympy\sympy\integrals\integrals.py in doit(self, **hints)
    528                 else:
    529                     antideriv = self._eval_integral(
--> 530                         function, xab[0], **eval_kwargs)
    531                     if antideriv is None and meijerg is True:
    532                         ret = try_meijerg(function, xab)

F:\Git\sympy\sympy\integrals\integrals.py in _eval_integral(self, f, x, meijerg, risch, manual, conds)
    989                 # rewrite using G functions
    990                 try:
--> 991                     h = meijerint_indefinite(g, x)
    992                 except NotImplementedError:
    993                     from sympy.integrals.meijerint import _debug

F:\Git\sympy\sympy\integrals\meijerint.py in meijerint_indefinite(f, x)
   1619     results = []
   1620     for a in sorted(_find_splitting_points(f, x) | {S(0)}, key=default_sort_key):
-> 1621         res = _meijerint_indefinite_1(f.subs(x, x + a), x)
   1622         if not res:
   1623             continue

F:\Git\sympy\sympy\integrals\meijerint.py in _meijerint_indefinite_1(f, x)
   1684         if b < 0 or f.subs(x, 0).has(nan, zoo):
   1685             place = None
-> 1686         r = hyperexpand(r.subs(t, a*x**b), place=place)
   1687
   1688         # now substitute back

F:\Git\sympy\sympy\simplify\hyperexpand.py in hyperexpand(f, allow_hyper, rewrite, place)
   2471         if not r.has(nan, zoo, oo, -oo):
   2472             return r
-> 2473     return f.replace(hyper, do_replace).replace(meijerg, do_meijer)

F:\Git\sympy\sympy\core\basic.py in replace(self, query, value, map, simultaneous, exact)
   1409             return expr
   1410
-> 1411         rv = bottom_up(self, rec_replace, atoms=True)
   1412
   1413         # restore original expressions for Dummy symbols

F:\Git\sympy\sympy\simplify\simplify.py in bottom_up(rv, F, atoms, nonbasic)
   1085             if args != rv.args:
   1086                 rv = rv.func(*args)
-> 1087             rv = F(rv)
   1088         elif atoms:
   1089             rv = F(rv)

F:\Git\sympy\sympy\core\basic.py in rec_replace(expr)
   1394             result = _query(expr)
   1395             if result or result == {}:
-> 1396                 new = _value(expr, result)
   1397                 if new is not None and new != expr:
   1398                     mapping[expr] = new

F:\Git\sympy\sympy\core\basic.py in <lambda>(expr, result)
   1337                 _value = lambda expr, result: value(*expr.args)
   1338             elif callable(value):
-> 1339                 _value = lambda expr, result: value(*expr.args)
   1340             else:
   1341                 raise TypeError(

F:\Git\sympy\sympy\simplify\hyperexpand.py in do_meijer(ap, bq, z)
   2468     def do_meijer(ap, bq, z):
   2469         r = _meijergexpand(G_Function(ap[0], ap[1], bq[0], bq[1]), z,
-> 2470                    allow_hyper, rewrite=rewrite, place=place)
   2471         if not r.has(nan, zoo, oo, -oo):
   2472             return r

F:\Git\sympy\sympy\simplify\hyperexpand.py in _meijergexpand(func, z0, allow_hyper, rewrite, place)
   2353         op._poly = Poly(op._poly.subs({z: 1/t, _x: -_x}), _x)
   2354     slater2, cond2 = do_slater(tr(func.bm), tr(func.an), tr(func.bq), tr(func.ap),
-> 2355                                t, 1/z0)
   2356
   2357     slater1 = powdenest(slater1.subs(z, z0), polar=True)

F:\Git\sympy\sympy\simplify\hyperexpand.py in do_slater(an, bm, ap, bq, z, zfinal)
   2279                 premult = (t/k)**bh
   2280                 hyp = _hyperexpand(Hyper_Function(nap, nbq), harg, ops,
-> 2281                                    t, premult, bh, rewrite=None)
   2282                 res += fac * hyp
   2283             else:

F:\Git\sympy\sympy\simplify\hyperexpand.py in _hyperexpand(func, z, ops0, z0, premult, prem, rewrite)
   2036
   2037     # We need to find the operators that convert formula into func.
-> 2038     ops += devise_plan(func, formula.func, z0)
   2039
   2040     # Now carry out the plan.

F:\Git\sympy\sympy\simplify\hyperexpand.py in devise_plan(target, origin, z)
   1613
   1614             if nbk[0] - namax <= 0 or bk[0] - amax <= 0:
-> 1615                 raise ValueError('Non-suitable parameters.')
   1616
   1617             if namax - amax > 0:

ValueError: Non-suitable parameters.
```

### LINKED ISSUE #17119: Integral fails to evaluate
Tested with sympy 1.4, issue repros on https://live.sympy.org/

```integrate(x**(0.5)*(1+x))```
crashes with ValueError: Non-suitable parameters.

The following both evaluate:
```integrate(x**(1/2)*(1+x))```
```integrate( expand(x**0.5*(1+x)) )```

This is the result from live.sympy.org:
```
>>> integrate(x**(0.5)*(1+x))
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/integrals/integrals.py", line 1477, in integrate
    return integral.doit(**doit_flags)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/integrals/integrals.py", line 541, in doit
    function, xab[0], **eval_kwargs)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/integrals/integrals.py", line 1012, in _eval_integral
    h = meijerint_indefinite(g, x)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/integrals/meijerint.py", line 1621, in meijerint_indefinite
    res = _meijerint_indefinite_1(f.subs(x, x + a), x)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/integrals/meijerint.py", line 1686, in _meijerint_indefinite_1
    r = hyperexpand(r.subs(t, a*x**b), place=place)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/simplify/hyperexpand.py", line 2491, in hyperexpand
    return f.replace(hyper, do_replace).replace(meijerg, do_meijer)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/core/basic.py", line 1483, in replace
    rv = bottom_up(self, rec_replace, atoms=True)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/simplify/simplify.py", line 1087, in bottom_up
    rv = F(rv)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/core/basic.py", line 1468, in rec_replace
    new = _value(expr, result)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/core/basic.py", line 1417, in <lambda>
    _value = lambda expr, result: value(*expr.args)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/simplify/hyperexpand.py", line 2488, in do_meijer
    allow_hyper, rewrite=rewrite, place=place)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/simplify/hyperexpand.py", line 2373, in _meijergexpand
    t, 1/z0)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/simplify/hyperexpand.py", line 2299, in do_slater
    t, premult, bh, rewrite=None)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/simplify/hyperexpand.py", line 2053, in _hyperexpand
    ops += devise_plan(func, formula.func, z0)
  File "/base/data/home/apps/s~sympy-live-hrd/57.417397408462217066/sympy/sympy/simplify/hyperexpand.py", line 1630, in devise_plan
    raise ValueError('Non-suitable parameters.')```

### PR DESCRIPTION
**Pull request by Smit-create**
Handling Float Integrals
#### References to other Issues or PRs

Fixes #17119
Fixes #14431

#### Brief description of what is fixed or changed
Before:
```
>>> integrate(x**(0.5)*(1+x))
Traceback (most recent call last):
  ValueError: Non-suitable parameters.

>>> integrate((x-t)**(-1/2)*t, (t,0,x))
 Traceback (most recent call last):
  ValueError: Non-suitable parameters.
```

After fixing:
```
>>> integrate(x**(0.5)*(1+x))
0.666666666666667*x**1.5 + 0.4*x**2.5

>>> integrate((x-t)**(-1/2)*t, (t,0,x))
1.33333333333333*x**1.5
```
#### Release Notes

* integrals
  * Handling Float Integrals

### CODE REVIEW

**Comment by asmeurer** [comment]
This would almost certainly break existing integrals that work. integrate can't always take an expression with a numeric parameter replaced with a symbolic one and still give a result. Sometimes a more general answer doesn't even exist.  It would be better to fix the underlying algorithm, in this case, meijerg, to work with floating point numbers.

**Comment by sylee957** [comment]
But whatever, it looks like there are only a few fixes to be done.

If it does not work (because using symbols in some places like exponents make the problem more difficult), then a way of casting float to rationals and casting back can be suggested. (Though I'm not sure that it is totally a good idea.)

**Comment by czgdp1807** [comment]
@Smit-create Are you still working on it? Please resolve the conflicts and address the comments.

**Comment by Smit-create** [comment]
Actually, I tried to fix it but still unable to find a proper fix yet. Would changing some tests work good?

**Comment by Smit-create** [comment]
Please review.

File: sympy/integrals/tests/test_meijerint.py [hunk 1]
```diff
@@ -709,7 +709,7 @@ def test_issue_10681():
     f = integrate(r**2*(R**2-r**2)**0.5, r, meijerg=True)
     g = (1.0/3)*R**1.0*r**3*hyper((-0.5, Rational(3, 2)), (Rational(5, 2),),
                                   r**2*exp_polar(2*I*pi)/R**2)
-    assert RR.almosteq((f/g).n(), 1.0, 1e-12)
+    assert RR.almosteq(simplify((f/g).n()), 1.0, 1e-12)
```
Comment (czgdp1807) [inline sympy/integrals/tests/test_meijerint.py]: Can we use `.simplify()` here?

**Comment by czgdp1807** [comment]
LGTM.

**Comment by oscarbenjamin** [comment]
This PR made `test_issue_15494` become a lot slower. Previously it took 20s but after this PR it takes 160s.

**Comment by Smit-create** [comment]
I will have a look at this.

**Comment by sylee957** [comment]
Actually I think that this changes are starting to fail in master because of timeouts.
I can consider reverting this patch if no idea is coming up from this.

**Comment by Smit-create** [comment]
If that is the case, then I suggest it to revert. As soon as, I find the other way around, I will try to fix this with as new PR. I do agree with you

**Comment by sylee957** [comment]
I didn’t know how much changes were done in master after this, but I see even reverting PR fails some test.

If the problem is about float integrals, I think that it’s better to test out to use nsimplify and cast floats into rationals, and restore them back.

**Comment by Smit-create** [comment]
> If the problem is about float integrals, I think that it’s better to test out to use nsimplify and cast floats into rationals, and restore them back.

I am thinking of a similar way to approach it.

**Comment by Smit-create** [comment]
```
diff --git a/sympy/integrals/integrals.py b/sympy/integrals/integrals.py
index 7cde17a290..c4a70638b2 100644
--- a/sympy/integrals/integrals.py
+++ b/sympy/integrals/integrals.py
@@ -28,6 +28,7 @@
 from sympy.series.order import Order
 from sympy.series.formal import FormalPowerSeries
 from sympy.simplify.fu import sincos_to_sum
+from sympy.simplify import nsimplify
 from sympy.utilities.misc import filldedent
 from sympy.utilities.exceptions import SymPyDeprecationWarning

@@ -398,9 +399,10 @@ def doit(self, **hints):


         if self.has(Float):
             fvals = self.atoms(Float)
-            fsubs = {f: Dummy() for f in fvals}
-            rsubs = {s: f for f, s in fsubs.items()}
-            return self.subs(fsubs).doit(**hints).subs(rsubs)
+            fsubs = {f: nsimplify(f) for f in fvals}
+            rresult_num, rresult_den = self.subs(fsubs).doit(**hints).as_numer_denom()
+            return rresult_num.evalf()/rresult_den.evalf()
```
The above diff leads some small test failures, but makes the `test_issue_15494` faster. Can you please help if there is some better way than this to handle this?

**Pull request merged**